hooks/terraform_docs.sh

```diff
@@ -40,8 +40,25 @@ function main {
 function replace_old_markers {
   local -r file=$1
 
-  sed -i "s/^${old_insertion_marker_begin}$/${insertion_marker_begin}/" "$file"
-  sed -i "s/^${old_insertion_marker_end}$/${insertion_marker_end}/" "$file"
+  # Detect sed version, if on MacOS use gsed to support -i parameter
+  if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  	SED_COMMAND="sed"
+  elif [[ "$OSTYPE" == "darwin"* ]]; then
+  	set +e # next command may fail - allow failure
+  	PRG="$(command -v "gsed" 2>/dev/null)"
+  	set -e # exit script on any error again
+  	if [ -z "$PRG" ]; then
+  		echo "ERROR: Detected MacOS but no gsed installed - install it via brew (brew install gsed)"
+  		exit 1
+  	fi
+  	SED_COMMAND="gsed"
+  else
+  	echo "ERROR: Unsupported OS system - hook supports MacOS and Linux only"
+  	exit 1
+  fi
+
+  ${SED_COMMAND} -i "s/^${old_insertion_marker_begin}$/${insertion_marker_begin}/" "$file"
+  ${SED_COMMAND} -i "s/^${old_insertion_marker_end}$/${insertion_marker_end}/" "$file"
```

Comment:
We can use POSIX `sed` on both platforms:
```suggestion
  sed -i'' -e "s/^${old_insertion_marker_begin}$/${insertion_marker_begin}/" "$file"
  sed -i'' -e "s/^${old_insertion_marker_end}$/${insertion_marker_end}/" "$file"
```